docarray/utils/lookup.py

```diff
@@ -0,0 +1,290 @@
+"""
+
+Originally from https://github.com/naiquevin/lookupy
```

Comment:
why do we copy/paste this instead of depending on it?

Reply:
we have some edits to do